homeassistant/setup.py

```diff
@@ -240,12 +242,20 @@ def log_error(msg: str) -> None:
     elif name in processed:
         return
 
-    if hasattr(module, 'DEPENDENCIES'):
-        dep_success = await _async_process_dependencies(
-            hass, config, name, module.DEPENDENCIES)  # type: ignore
+    manifest = await hass.async_add_executor_job(
+        json_util.load_json,
+        os.path.join(os.path.dirname(module.__file__), 'manifest.json')
+    )
+
+    deps = (manifest['dependencies'] if manifest else getattr(module, 'DEPENDENCIES', []))  # type: ignore # noqa: line-to-long # pylint: disable=line-too-long
```

Comment:
This is really ugly, but I couldn't get mypy to allow `manifest['dependencies'] with this as a multiline statement. Unfortunately, flake8/pylint had to be overridden because the line is now way too long.